index.bs

```diff
@@ -3701,6 +3727,792 @@ opened</dfn> steps given |window|, |type| and |message|.
+   timing|'s [=connection start time=] and |global|.
+
+1. Let |connect end| be [=convert fetch timestamp=] given |connection timing|'s
+   [=connection end time=] and |global|.
+
+1. Let |request start| be [=convert fetch timestamp=] given |timings|' [=final
+   network-request start time=] and |global|.
+
+1. Let |response start| be [=convert fetch timestamp=] given |timings|' [=final
+   network-response start time=] and |global|.
+
+1. Let |response end| be [=convert fetch timestamp=] given |timings|'
+   <a spec=fetch>end time</a> and |global|.
+
+1. Return a [=map=] matching the <code>network.FetchTimingInfo</code> production
+    with the <code>requestTime</code> field set to |request time|, the
+    <code>redirectStart</code> field set to |redirect start|, the
+    <code>redirectEnd</code> field set to |redirect end|, the
+    <code>fetchStart</code> field set to |fetch start|, the
+    <code>dnsStart</code> field set to |DNS start|, the <code>dnsEnd</code>
+    field set to |DNS end|, the <code>connectStart</code> field set to |connect
+    start|, the <code>connectEnd</code> field set to |connect end|, the
+    <code>tlsStart</code> field set to |TLS start|, the
+    <code>requestStart</code> field set to |request start|, the
+    <code>responseStart</code> field set to |response start|, and the
+    <code>responseEnd</code> field set to |response end|.
+
+</div>
+
+TODO: Add service worker fields
+
+#### The network.Header type #### {#type-network-Header}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.Header = {
+    name: text,
+    ? value: text,
+    ? binaryValue: [ uint ]
+};
+</pre>
+
+The <code>network.Header</code> type represents a single request header.
+
+If the header value can be represented as a UTF-8 encoded string, the
+<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
+field will be present and consist of an array of integers representing the bytes
+of the header.
+
+<div algorithm>
+To <dfn>get a header</dfn> given |name bytes| and |value bytes|:
+
+1. Let |name| be the result of [=UTF-8 decode=] with |name bytes|.
+
+   Assert: Since header names are constrained to be ASCII-only this cannot fail.
+
+1. Let |utf8 decoded value| be the result of [=UTF-8 decode without BOM or fail=]
+   with |value bytes|.
+
+1. If |utf8 decoded value| is failure, then:
+
+    1. Let |value| be null and |binary value| be an empty list.
+
+    1. For each |byte| in |value bytes|:
+
+      1. Append the [=byte/value=] of |byte| to |binary value|.
+
+   Otherwise: let |value| be |utf8 decoded value| and let |binary value|
+   be null.
+
+1. Return a map matching the <code>network.Header</code> production, with the
+   <code>name</code> field set to |name|, the <code>value</code> field
+   set to |value| if it's not null, or omitted otherwise, and the
+   <code>binaryValue</code> field set to |binary value| if it's not null, or
```

Comment:
_binary value_ might not have been initialized here. Maybe let _value_ and _binary value_ be null first, and only set one of them?